mitreattack/navlayers/generators/overview_generator.py

```diff
@@ -4,68 +4,70 @@
     from ..exporters.matrix_gen import MatrixGen
     from ..core.exceptions import BadInput, typeChecker, categoryChecker
     from ..core.layer import Layer
+    from ..generators.gen_helpers import remove_revoked, construct_relationship_mapping
 except ValueError:
     from mitreattack.navlayers.exporters.matrix_gen import MatrixGen
     from mitreattack.navlayers.core.exceptions import BadInput, typeChecker, categoryChecker
     from mitreattack.navlayers.core.layer import Layer
+    from mitreattack.navlayers.generators.gen_helpers import remove_revoked, construct_relationship_mapping
 except ImportError:
     from navlayers.exporters.matrix_gen import MatrixGen
     from navlayers.core.exceptions import BadInput, typeChecker, categoryChecker
     from navlayers.core.layer import Layer
+    from navlayers.generators.gen_helpers import remove_revoked, construct_relationship_mapping
 
 
 class UnableToFindTechnique(Exception):
     pass
 
 
 class OverviewGenerator:
-
+    """Generates a Layer file that provides an overview of entities related to each technique"""
     def __init__(self, source, matrix='enterprise', local=None):
+        """
+        Initialize the Generator
+        :param source: Which source to use for data (local or taxii [server])
+        :param matrix: Which matrix to use during generation
+        :param local: Optional path to local data
+        """
         self.matrix_handle = MatrixGen(source, local)
         self.domain = matrix
         try:
             self.source_handle = self.matrix_handle.collections[matrix]
         except KeyError:
             print(f"[UsageGenerator] - unable to load collection {matrix} (current source = {source}).")
             raise BadInput
-        self.tech_listing = self.remove_revoked(self.source_handle.query([Filter('type', '=', 'attack-pattern')]))
-        self.group_objects = self.source_handle.query([Filter('type', '=', 'course-of-action')])
+        tl = remove_revoked(self.source_handle.query([Filter('type', '=', 'attack-pattern')]))
+        self.mitigation_objects = self.source_handle.query([Filter('type', '=', 'course-of-action')])
         complete_relationships = self.source_handle.query([Filter('type', '=', 'relationship'),
                                                            Filter('relationship_type', '=', 'uses')])
         complete_relationships.extend(self.source_handle.query([Filter('type', '=', 'relationship'),
                                                                 Filter('relationship_type', '=', 'mitigates')]))
+        # Contains relationship mapping [stix id] -> [relationships associated with that stix id for each type]
         self.mitigation_relationships = {}
         self.software_relationships = {}
         self.group_relationships = {}
```

Comment:
Yeah, that's a reasonable simplification at this point.

Reply:
To clarify, this wasn't actually dead code - the individual dictionaries were reference later on, as simplifier was just a handle too them. I've reworked things to make everything reference simplifier now, rather than the individual dicts.